### PR TITLE
Display cover image thumbnail on compact post display

### LINKED ISSUE #1595: Article: display cover image as thumbnail on post summary card

### PR DESCRIPTION
**Pull request by mbertrand**
#### Pre-Flight checklist
- [x] Screenshots and design review for any changes that affect layout or styling
  - [x] Desktop screenshots 
  - [x] Mobile width screenshots 
  - [x] tag @ferdi or @pdpinch for review  
- [x] Testing
  - [x] Code is tested
  - [x] Changes have been manually tested

#### What are the relevant tickets?
Closes #1595

#### What's this PR do?
- Modifies `CompactPostDisplay` component to display the cover image thumbnail for an article if present.
- On the backend, crops the image before creating the article cover thumbnail to mimic embedly's API for external link-post thumbnails.

#### How should this be manually tested?
- Add an article with cover image via django shell:
```python
import os
import json
from django.contrib.auth.models import User
from django.urls import reverse
from rest_framework.test import APIClient
from channels.models import Channel, Post, Article

client = APIClient()
client.force_login(User.objects.get(username='<username>'))
channel = Channel.objects.get(name="<channel_name>")
article_content = [{"key": "value", "nested": {"number": 4}}]
img_file = "static/images/avatar_default.png"
url = reverse("post-list", kwargs={"channel_name": channel.name})
with open(img_file, "rb") as f:
    resp = client.post(
        url, 
        {"title": "foo", "article_content": json.dumps(article_content), "cover_image": f}, 
        format="multipart"
    )

post = Post.objects.get(post_id=resp.json()["id"])
print(post.article.cover_image_small.url)
 ```
- Go to the article's channel page.  The thumbnail should be visible and have the same dimensions as a link post thumbnail.

#### Screenshots (if appropriate)
Two link-posts and two articles were created using the same images for thumbnails.  

**Desktop channel page**
screen shot 2018-12-27 at 11 21 12 am

**Desktop frontpage - wide**
screen shot 2018-12-27 at 12 05 32 pm

**Mobile:**
screen shot 2018-12-27 at 11 21 42 am

### CODE REVIEW

**Comment by mbertrand** [comment]
@Ferdi screenshots attached

File: channels/utils.py [hunk 1]
```diff
@@ -19,7 +19,7 @@
 AVATAR_SMALL_MAX_DIMENSION = 22
 AVATAR_MEDIUM_MAX_DIMENSION = 90
 
-COVER_IMAGE_THUMBNAIL_DIMENSION = 201
+COVER_IMAGE_THUMBNAIL_DIMENSIONS = (201, 103)
```
Comment (rhysyngsun) [inline channels/utils.py]: I would suggest making this a `namedtuple`.

File: profiles/utils.py [hunk 2]
```diff
@@ -263,6 +263,42 @@ def make_thumbnail(full_size_image, max_dimension):
...
+            A file-like object containing an image. This file will seek back to the beginning after being read.
+        max_dimensions (int, int):
+            The max width and height for the thumbnail
+    Returns:
+        BytesIO:
+            A jpeg image which is a thumbnail of full_size_image
+    """
+    pil_image = Image.open(full_size_image)
+    x, y = max_dimensions
+    if pil_image.height / pil_image.width < y / x:
```
Comment (rhysyngsun) [inline profiles/utils.py]: suggest refactoring `y / x` to an `aspect_ratio` var to make this a little cleaner

File: static/js/components/CompactPostDisplay.js [hunk 3]
```diff
@@ -122,28 +122,45 @@ export class CompactPostDisplay extends React.Component<Props> {
...
+                          post.thumbnail,
+                          103,
+                          201
+                        )}
+                      />
+                    </a>
+                  ) : null}
+                </React.Fragment>
+              ) : post.thumbnail ? (
+                <img src={post.thumbnail} />
```
Comment (rhysyngsun) [inline static/js/components/CompactPostDisplay.js]: I think this image should be linked too?

File: static/js/components/CompactPostDisplay.js [hunk 4]
```diff
@@ -122,28 +122,45 @@ export class CompactPostDisplay extends React.Component<Props> {
...
+                    >
+                      <i className="material-icons open_in_new top-right overlay-icon">
+                        open_in_new
+                      </i>
+                    </a>
+                  </div>
+                  {post.thumbnail ? (
+                    <a
+                      href={
+                        post.url ||
```
Comment (rhysyngsun) [inline static/js/components/CompactPostDisplay.js]: `post.url` is always truthy here now

File: static/js/components/CompactPostDisplay.js (hunk 3 above)
Comment (mbertrand) [reply static/js/components/CompactPostDisplay.js]: [Ferdi said it should behave like a text post](https://github.com/mitodl/open-discussions/issues/1595#issuecomment-450193184), so I left out the link.

**Pull request merged**